### ISSUE TITLE
Globe does not use shortest path for line geometry between points

### ISSUE THREAD

**Issue by karimnaaji** [mapbox/mapbox-gl-js#11813]
Line geometry between points on globe should probably use shortest distance and be preprocess in geojson-vt, there's also non-intuitive behavior around the poles where the points are projected in mercator space. The cheap-ruler might be useful to fix that issue: https://github.com/mapbox/cheap-ruler. Refer:

https://user-images.githubusercontent.com/7061573/164815481-3ce73cb1-7364-4055-98c2-92ab9d3befd5.mov

**Comment by mootari** [comment]
This also seems to affect `flyTo` transitions between two points, which follow the curves described above instead of a straight line.

**Comment by runfalk** [comment]
I ran into this exact issue when trying to plot a flight path across the north pole. It lead to some very surprising interactions where I needed to manually inject points close to true north width longitudes 34 and 192 to prevent it from going completely haywire.

I still ended up with this abomination at the pole:

![bild](https://github.com/mapbox/mapbox-gl-js/assets/864128/6c06d964-dfa5-44a0-b445-63a72211c74f)

This should just be two dotted lines.

**Comment by blumk** [comment]
Issue still exists in `v3.0.1`

### Example
Plotting line between `San Francisco` and `Auckland` does not render shortest path:

![image](https://github.com/mapbox/mapbox-gl-js/assets/2436496/7e3f2ff8-ef7f-4bf6-aa80-5da8f24d2f7c)

```
    <Map
      mapLib={import("mapbox-gl")}
      initialViewState={{
        longitude: -100,
        latitude: 40,
        zoom: 1,
      }}
      style={{ width: "100vw", height: "100vh" }}
      mapStyle="mapbox://styles/blumk/clqrgy8cf00cu01ob4nrufv5k"
      mapboxAccessToken={mapboxToken}
      onRender={(event) => event.target.resize()}
    >
      <Source
        id="polylineLayer"
        type="geojson"
        data={{
          type: "Feature",
          properties: {},
          geometry: {
            type: "LineString",
            coordinates: [
              [174.7917, -37.008],
              [-122.375, 37.6188],
            ],
          },
        }}
      >
        <Layer
          id="lineLayer"
          type="line"
          source="my-data"
          layout={{
            "line-join": "round",
            "line-cap": "round",
          }}
          paint={{
            "line-color": "rgba(3, 170, 238, 0.5)",
            "line-width": 5,
          }}
        />
      </Source>
    </Map>
```

**Comment by mklnz** [comment]
Having the same issue but from Auckland to New York, it doesn't go across the pacific.

**Comment by LukasJK** [comment]
> I ran into this exact issue when trying to plot a flight path across the north pole. It lead to some very surprising interactions where I needed to manually inject points close to true north width longitudes 34 and 192 to prevent it from going completely haywire.
> 
> I still ended up with this abomination at the pole:
> 
> ![bild](https://private-user-images.githubusercontent.com/864128/264366095-6c06d964-dfa5-44a0-b445-63a72211c74f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NDY5NDgsIm5iZiI6MTcyMTc0NjY0OCwicGF0aCI6Ii84NjQxMjgvMjY0MzY2MDk1LTZjMDZkOTY0LWRmYTUtNDRhMC1iNDQ1LTYzYTcyMjExYzc0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QxNDU3MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jM2M1OTU3YjVmYTE4MGZiNjMxYjMyMmU5YzcyNDdhNjk1NWQ5Njg5NTRiODEyNWE4OTE0MmQ3ODUzNWYyMWViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OsVABlmOJqcVKRGg32pHCMoKvgzen12QXB7rKgJGf-c)
> 
> This should just be two dotted lines.

Were you ever able to come up with some workaround?

**Comment by runfalk** [comment]
Nope, that ugly blob was with my workaround of injecting points close to the pole.

**Comment by QuentinDEME97** [comment]
Hey ! I also face this issue ! 🥲

My dots should be connected but it goes all around the world to connect...

<img width="1098" alt="image" src="https://github.com/user-attachments/assets/2d3527a2-8ba2-4308-a600-14cf9fe1be9f">
